src/phoenix/trace/llama_index/callback.py

```diff
@@ -18,21 +17,43 @@
 
 
 def _check_instrumentation_compatibility() -> bool:
-    if find_spec("llama_index") is None:
-        raise PackageNotFoundError("Missing `llama-index`. Install with `pip install llama-index`.")
-    # split the version string into a tuple of integers
-    llama_index_version_str = version("llama-index")
-    llama_index_version = tuple(map(int, llama_index_version_str.split(".")[:3]))
+    try:
+        llama_index_version_str = version("llama-index")
+    except PackageNotFoundError:
+        llama_index_version_str = None
+
+    try:
+        llama_index_core_version_str = version("llama-index-core")
+    except PackageNotFoundError:
+        llama_index_core_version_str = None
+
     instrumentation_version_str = version("openinference-instrumentation-llama-index")
-    instrumentation_version = tuple(map(int, instrumentation_version_str.split(".")[:3]))
-    # check if the llama_index version is compatible with the instrumentation version
+    instrumentation_version = _parse_semantic_version(instrumentation_version_str)
+
+    if llama_index_version_str is None:
+        if llama_index_core_version_str is None:
+            raise PackageNotFoundError(
+                "Missing `llama_index`. "
+                "Install with `pip install llama-index` or "
+                "`pip install llama-index-core` for a minimal installation."
+            )
+        if instrumentation_version < INSTRUMENTATION_MODERN_VERSION:
+            raise IncompatibleLibraryVersionError(
+                f"llama-index-core v{llama_index_core_version_str} is not compatible with "
+                f"openinference-instrumentation-llama-index v{instrumentation_version_str}. "
+                "Please upgrade openinference-instrumentation-llama-index to at least 1.0.0 via "
+                "`pip install 'openinference-instrumentation-llama-index>=1.0.0'`."
+            )
```

Comment:
maybe I'm reading the indentation wrong but I feel this is in the wrong place?

Reply:
This condition is checking for the case where `llama-index-core` is installed but `llama-index` is not installed. In that case, I am checking whether the current version of the instrumentation is compatible.

Reply:
I see, yeah the conditionals a a bit confusing. I think it may be good to explain your logic here a bit as it's quite confusing to map the conditionals